src/Core/Cart/Calculator.php

```diff
@@ -200,7 +200,15 @@ public function getRowTotalWithoutDiscount()
     public function getDiscountTotal()
     {
         $amount = new AmountImmutable();
+        $isFreeShippingAppliedToAmount = false;
         foreach ($this->cartRules as $cartRule) {
+            if ((bool) $cartRule->getRuleData()['free_shipping']) {
+                if ($isFreeShippingAppliedToAmount) {
+                    $initialShippingFees = $this->getFees()->getInitialShippingFees();
+                    $amount = $amount->sub($initialShippingFees);
+                }
```

Comment:
```suggestion
                if (!$isFreeShippingAppliedToAmount) {
                    $initialShippingFees = $this->getFees()->getInitialShippingFees();
                    $amount = $amount->sub($initialShippingFees);
                }
```

This must be applied only the first time when the variable is defined at `false`